### PR TITLE
Issue#32   spi

### PR DESCRIPTION
**Pull request by oallaire**

### CODE REVIEW

File: library/src/hspec/stm/stm32f0/mal_hspec_stm32f0_spi.h [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+#include "std/mal_error.h"
+#include "hspec/mal_hspec_cmn.h"
+#include "stm32f0/stm32f0xx_misc.h"
+#include "std/mal_bool.h"
+
+#define mal_hspec_stm32f0_spi_enable_interrupt(interface, active) do {\
+	if (active) { \
+		NVIC_EnableIRQ(mal_hspec_stm32f0_spi_get_irq(interface)); \
+	} \
+} while(0)
```
Comment (larosek) [inline library/src/hspec/stm/stm32f0/mal_hspec_stm32f0_spi.h]: Maybe I'm missing something about macro so sorry if this is an obvious question. Why do we need to do a do while(0)? Why not just the if alone?

**Comment by larosek** [comment]
Other than my question about the macro with a do while(0) it all looks good :)

File: library/src/hspec/stm/stm32f0/mal_hspec_stm32f0_spi.h (hunk 1 above)
Comment (oallaire) [inline library/src/hspec/stm/stm32f0/mal_hspec_stm32f0_spi.h]: It's a convention for macros when it should not be used other then a one liner. Putting the code inside a code block ({}) tells the reader and the compiler that this code is not be used inside a condition check or as an operand.

The do while(0) is the convention so any macro that I want to outline as one liner, I put in a do while(0).

TLDR: Convention :P
